pkgs/games/hmcl-bin/default.nix

```
@@ -0,0 +1,88 @@
+{ lib
+, stdenv
+, file
+, glib
+, fetchurl
+, makeWrapper
+, wrapGAppsHook
+, jdk
+, alsa-lib
+, xorg
+, wayland
+, libpulseaudio
+, libglvnd
+, libGL
+, dconf
+, glfw
+, makeDesktopItem
+, copyDesktopItems
+, type ? "jar"
+, sourceProvenance ? (if type == "jar" then [ lib.sourceTypes.binaryBytecode ] else [])
+, openal }:
+
+stdenv.mkDerivation rec {
+  name = "hmcl-bin";
+  version = "3.5.3.223";
+
+  src = fetchurl {
+    url = "https://github.com/huanghongxun/HMCL/releases/download/v${version}/HMCL-${version}.jar";
+    sha256 = "sha256-8g2FMvAiAKfxJUY0G7wl6d44wOpVsknFHGZ85IkOzFc=";
+  };
+
+  dontUnpack = true;
+
+  icon = fetchurl {
+    url = "https://aur.archlinux.org/cgit/aur.git/plain/craft_table.png?h=hmcl-bin";
+    sha256 = "sha256-KYmhtTAbjHua/a5Wlsak5SRq+i1PHz09rVwZLwNqm0w";
+  };
+
+  buildInputs = [ (lib.getLib dconf) glib ];
+  nativeBuildInputs = [ jdk wrapGAppsHook makeWrapper file copyDesktopItems ];
+
+  installPhase = let
+    libpath = with xorg; lib.makeLibraryPath ([
+      libGL
+      glfw
+      openal
+      libglvnd
+    ] ++ lib.lists.optionals stdenv.isLinux [
+      libX11
+      libXext
+      libXcursor
+      libXrandr
+      libXxf86vm
+      libpulseaudio
+      wayland
+    ]);
+  in ''
+    runHook preInstall
+    mkdir -p $out/{bin,lib/hmcl-bin}
+    cp $src $out/lib/hmcl-bin/hmcl-bin.jar
+    install -Dm644 $icon $out/share/icons/hicolor/48x48/apps/hmcl.png
+    makeWrapper  ${jdk}/bin/java $out/bin/hmcl-bin \
```

Did not test, since you also added wrapGAppsHook, will this result in double wrapping? Can we just pass gappsWrapperArgs here and `dontWrapGApps = true;`?

---

```suggestion
    makeWrapper ${jdk}/bin/java $out/bin/hmcl-bin \
```
We are not using the gapps wrapper args anywhere?